src/server/scripts/Northrend/VioletHold/boss_moragg.cpp

```diff
@@ -134,7 +147,149 @@ class boss_moragg : public CreatureScript
     }
 };
 
+class spell_moragg_ray_of_suffering : public SpellScriptLoader
+{
+public:
+    spell_moragg_ray_of_suffering() : SpellScriptLoader("spell_moragg_ray_of_suffering") { }
+
+    class spell_moragg_ray_of_suffering_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_moragg_ray_of_suffering_AuraScript);
+
+        void OnPeriodic(AuraEffect const* aurEff)
+        {
+            PreventDefaultAction();
+            std::list<HostileReference*> players = GetTarget()->ToCreature()->getThreatManager().getThreatList();
```

Comment:
Shouldnt be something like... without ToCreature()? i mean we want to select players or players and creatures (pets)? normaly GetTarget() return a unit type so it should work well... correct me if i'm wrong

std::list<HostileReference*> players = GetTarget()->getThreatManager().getThreatList();